packages/synapse-interface/components/Portfolio/Portfolio.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useAccount, useNetwork } from 'wagmi'
+import { Zero } from '@ethersproject/constants'
+import { RootState } from '@/store/store'
+import { setFromChainId } from '@/slices/bridgeSlice'
+import { PortfolioTabManager } from './PortfolioTabManager'
+import {
+  usePortfolioBalancesAndAllowances,
+  NetworkTokenBalancesAndAllowances,
+  TokenWithBalanceAndAllowance,
+  FetchState,
+} from '@/utils/hooks/usePortfolioBalances'
+import { PortfolioContent, HomeContent } from './PortfolioContent'
+
+export enum PortfolioTabs {
+  HOME = 'home',
+  PORTFOLIO = 'portfolio',
+}
+
+export const Portfolio = () => {
+  const [tab, setTab] = useState<PortfolioTabs>(PortfolioTabs.HOME)
+
+  const dispatch = useDispatch()
+  const { fromChainId, bridgeTxHashes } = useSelector(
+    (state: RootState) => state.bridge
+  )
+
+  const {
+    balancesAndAllowances: portfolioData,
+    fetchPortfolioBalances,
+    status: fetchState,
+  } = usePortfolioBalancesAndAllowances()
+
+  const filteredPortfolioDataForBalances: NetworkTokenBalancesAndAllowances =
+    filterPortfolioBalancesWithBalances(portfolioData)
+
+  const { address } = useAccount()
+  const { chain } = useNetwork()
+
+  useEffect(() => {
+    if (address) {
+      setTab(PortfolioTabs.PORTFOLIO)
+    }
+  }, [address])
+
+  useEffect(() => {
+    ;(async () => {
+      if (address && chain.id) {
+        await dispatch(setFromChainId(chain.id))
+        await fetchPortfolioBalances()
+      }
+    })()
+  }, [address, chain])
+
+  return (
+    <div
+      data-test-id="portfolio"
+      className="flex flex-col w-full max-w-lg mx-auto lg:mx-0"
+    >
+      <PortfolioTabManager activeTab={tab} setTab={setTab} />
+      <div className="mt-4">
+        {tab === PortfolioTabs.HOME && <HomeContent />}
+        {tab === PortfolioTabs.PORTFOLIO && (
+          <PortfolioContent
+            connectedAddress={address}
+            connectedChainId={chain?.id}
+            selectedFromChainId={fromChainId}
+            networkPortfolioWithBalances={filteredPortfolioDataForBalances}
+            fetchPortfolioBalancesCallback={fetchPortfolioBalances}
+            fetchState={fetchState}
+            bridgeTxHashes={bridgeTxHashes}
+          />
+        )}
+      </div>
+    </div>
+  )
+}
+
+function filterPortfolioBalancesWithBalances(
```

Comment:
The filterPortfolioBalancesWithBalances function could be improved by using .filter() instead of .forEach() and re-creating the object. This would be more efficient:

```
function filterPortfolioBalancesWithBalances(balancesAndAllowances) {
  return Object.fromEntries(
    Object.entries(balancesAndAllowances).map(([key, tokenWithBalances]) => {
      const filtered = tokenWithBalances.filter(t => t.balance > Zero)  
      return [key, filtered]
    })
  )
}
```

Reply:
That's a great point! Using `.filter()` method would be more efficient than the latter. 

Made a slight adjustment to the code you provided - needed to add an additional `.filter()` call to filter out Portfolios that didn't hold balances (otherwise all networks would display, regardless of balance): 

`
function filterPortfolioBalancesWithBalances(
  balancesAndAllowances: NetworkTokenBalancesAndAllowances
): NetworkTokenBalancesAndAllowances {
  const filteredEntries = Object.entries(balancesAndAllowances).map(
    ([key, tokenWithBalances]) => {
      const filtered = tokenWithBalances.filter(
        (token: TokenWithBalanceAndAllowance) => token.balance > Zero
      )
      return [key, filtered]
    }
  )

  const filteredEntriesWithValues = filteredEntries.filter(
    ([_, filtered]) => filtered.length > 0
  )

  return Object.fromEntries(filteredEntriesWithValues)
}
`
Updated in commit https://github.com/synapsecns/sanguine/pull/1104/commits/260ff5296f20b56f65a9f989586177ecaa94cc8b

Reply:
Further optimizing function by using `.reduce()` method as opposed to `.map() + .filter()`: 

```
function filterPortfolioBalancesWithBalances(
  balancesAndAllowances: NetworkTokenBalancesAndAllowances
): NetworkTokenBalancesAndAllowances {
  return Object.entries(balancesAndAllowances).reduce(
    (
      filteredBalances: NetworkTokenBalancesAndAllowances,
      [key, tokenWithBalances]
    ) => {
      const filtered = tokenWithBalances.filter(
        (token: TokenWithBalanceAndAllowance) => token.balance > Zero
      )
      if (filtered.length > 0) {
        filteredBalances[key] = filtered
      }
      return filteredBalances
    },
    {}
  )
}
```

Updated in commit https://github.com/synapsecns/sanguine/pull/1104/commits/deef779f25ed6ad26245c01379f07ace728c10d7